src/index.ts

```diff
@@ -13,15 +20,27 @@ const hasFeatureSupport = hasProperties(window, requiredFeatures);
 const state: Fastly = {
   client: {
     hasFeatureSupport
-  }
+  },
+  // The are certain scenarios (such as the demo website) in which the gloabl
+  // FASTLY and the results array already exists.
+  results: (window.FASTLY && window.FASTLY.results) || []
 };
 
 // Init initializes the library when the browser is ready
 export function init(): void {
   loadWhenDocumentReady(
     (): void => {
       const params = getScriptParameters(SCRIPT_SRC_REGEXP);
-      worker.init(params).catch((): void => {});
+      worker
+        .init(params)
+        .then(
+          (results: Beacon[]): void => {
+            // We use apply here as we don't want to directly assign i.e.
+            // replace the existing results array.
+            Array.prototype.push.apply(state.results, results);
```

Comment:
Not sure I understand why you don't want to use `concat` or spread here? You are mutating the object so what's the diff between that and reassignment?

Reply:
The demo website works by binding a `new Proxy([], handler)` function to the property `window.FASTLY.results`, so whenever the target array that the proxy observes is read or written to we can display the results in the UI. Therefore, we cannot re-assign a new array (both `concat` and spread create new arrays) to the property as it will remove the proxy. Therefore we need to push into the results instead. Happy to explain in more depth offline :)